### PR TITLE
Transpilation JSLIGO

### PR DESCRIPTION
**Pull request by ghost**

### CODE REVIEW

File: src/contracts/jsligo/betting/errors.jsligo [hunk 1]
```diff
@@ -0,0 +1,53 @@
+// --------------------------------------
+//         CONFIG RELATED Errors
+// --------------------------------------
+
+const not_manager : string = "Not the Manager of the contract";
```
Comment (SanderSpies) [inline src/contracts/jsligo/betting/errors.jsligo:5]: Type annotations are not needed here.

File: src/contracts/jsligo/betting/main.jsligo [hunk 2]
```diff
@@ -0,0 +1,309 @@
+#import "types.jsligo" "Types"
+#import "errors.jsligo" "Errors"
+#import "assert.jsligo" "Assert"
+#import "callback/main.jsligo" "Callback"
+
+const changeManager = (pNewManager: address, s: Types.storage) => {
+  let _ = Assert.isManager (Tezos.get_sender(), s.manager);
```
Comment (SanderSpies) [inline src/contracts/jsligo/betting/main.jsligo]: `let _ = ` can be removed.

File: src/contracts/jsligo/oracle/main.jsligo [hunk 3]
```diff
@@ -0,0 +1,80 @@
+#import "types.jsligo" "Types"
+#import "assert.jsligo" "Assert"
+#import "errors.jsligo" "Errors"
+#import "callback/main.jsligo" "Callback"
+
+// --------------------------------------
+//      CONFIGURATION INTERACTIONS
+// --------------------------------------
+
+const changeManager = (pNewManager: address, s: Types.storage) : Types.return_ => {
```
Comment (SanderSpies) [inline src/contracts/jsligo/oracle/main.jsligo]: Return type can be removed.

File: test/betting/helpers/assert.jsligo [hunk 4]
```diff
@@ -0,0 +1,83 @@
...
+const string_failure = ([res, p_expected] : [test_exec_result, string]) : unit => {
+  let expected = Test.eval (p_expected) ;
+  match (res, {
+    Fail: (x: test_exec_error) => (
+      match (x, {
+        Rejected : (x : [michelson_code, address]) => assert (Test.michelson_equal (x[0], expected)),
+        Balance_too_low : (_: {contract_too_low : address, contract_balance : tez, spend_request : tez}) => Test.failwith ("contract failed for an unknown reason"),
+        Other : (_ : string) => Test.failwith ("contract failed for an unknown reason")
+      })),
+    Success: (_:nat) => Test.failwith ("bad price check")
```
Comment (SanderSpies) [inline test/betting/helpers/assert.jsligo]:
```suggestion
    Success: _ => Test.failwith ("bad price check")
```

File: test/betting/helpers/assert.jsligo [hunk 5]
```diff
@@ -0,0 +1,83 @@
...
+const manager = (ctr_taddr : typed_address<Types.action, Types.storage>, expected : address) : unit => {
+    let ctr_storage : Types.storage = Test.get_storage(ctr_taddr);
+    let ctr_value : address = (ctr_storage.manager);
+    assert (ctr_value == expected)
+};
+
+// Assert Oracle Address parameter with expected result 
+const oracle = ([ctr_taddr, expected] : [typed_address<Types.action, Types.storage>, address]) : unit => {
+    let ctr_storage : Types.storage = Test.get_storage(ctr_taddr);
+    let ctr_value : address = (ctr_storage.oracle_address);
```
Comment (SanderSpies) [inline test/betting/helpers/assert.jsligo]:
```suggestion
    let ctr_value = ctr_storage.oracle_address;
```

**Review by SanderSpies — COMMENTED** [review]
Overall looks good. Some small notes:
- remove the `let _ = ` lines
- return types for functions are often not needed
- types for literals/variables are often not needed

File: src/contracts/jsligo/betting/main.jsligo (hunk 2 above)
Comment (melwyn95) [reply src/contracts/jsligo/betting/main.jsligo]: Done

File: src/contracts/jsligo/oracle/main.jsligo (hunk 3 above)
Comment (melwyn95) [reply src/contracts/jsligo/oracle/main.jsligo]: Done

File: test/betting/helpers/assert.jsligo (hunk 4 above)
Comment (melwyn95) [reply test/betting/helpers/assert.jsligo]: Done

File: test/betting/helpers/assert.jsligo (hunk 5 above)
Comment (melwyn95) [reply test/betting/helpers/assert.jsligo]: Done

File: src/contracts/jsligo/betting/errors.jsligo (hunk 1 above)
Comment (melwyn95) [reply src/contracts/jsligo/betting/errors.jsligo:5]: Done, removed

**Review by SanderSpies — APPROVED** [review]

**Pull request merged**